src/app/data-analytics-dashboard/oncology/oncology-reports/oncology-report-pdf-view/oncology-report-pdf.service.ts

```diff
@@ -0,0 +1,501 @@
+    this.constructAggregateOuterLayout(data, params);
+    return Observable.create((observer: Subject<any>) => {
+      this.getLogo('./assets/img/ampath.png', (letterHead) => {
+        observer.next({
+          pageSize: 'LETTER',
+          pageMargins: 42,
+          footer: {
+            stack: [{
+              bold: true,
+              color: 'black',
+              text: ' Generated On: ' + new Date(),
+              style: { alignment: 'center', fontSize: 8 }
+            }
+            ],
+            margin: [42, 20]
+          },
+          content: [{
+            stack: [
+              {
+                image: letterHead,
+                width: 150,
+                alignment: 'center'
+              },
+              {
+                text: aggregatedData.locations,
+                margin: [0, 5, 0, 5],
+                alignment: 'center',
+                style: 'locationStyle'
+              },
+              {
+                text: this._formatReportIndicators(title) + ' Aggregate Report',
+                style: 'header',
+                alignment: 'center'
+              },
+              {
+                columns:
+                  [
+                    { width: '*', style: 'dateColumn', text: 'Start Date: ' + params.startDate },
+                    { width: '*', style: 'dateColumn', text: 'End Date: ' + params.endDate }
+                  ],
+                bold: true,
+                color: '#2a2a2a',
+                fillColor: '#d3d3d3'
+              },
+              aggregatedData.sections[0].body,
+            ]
+          },
+          {
+            layout: 'noBorders',
+            table: {
+              widths: ['*', '*'],
+              body: this.constructAggregateOuterLayout(data, params)
+            }
+          }
+          ],
+          styles: {
+            header: {
+              fontSize: 14,
+              bold: true,
+              margin: [0, 0, 0, 10]
+            },
+            locationStyle: {
+              fontSize: 12,
+              bold: true,
+              margin: [0, 10, 0, 5]
+            },
+            dateColumn: {
+              alignment: 'center',
+              fontSize: 10,
+              bold: true,
+            },
+            defaultTable: {
+              fontSize: 8,
+              margin: [0, 10, 0, 5]
+            },
+            headerstyle: {
+              fontSize: 8,
+              bold: true,
+              color: '#2a2a2a',
+              fillColor: '#d3d3d3'
+            },
+            aggTable: {
+              fontSize: 8,
+              margin: [0, 10, 0, 5]
+            }
+          },
+          defaultStyle: {
+            fontSize: 8
+          }
+        });
+      });
+    }).pipe(first());
+
+  }
+
+  public constructPdfSections(data: Array<any>, params: any): Array<Array<any>> {
+    const sectionsArray: Array<Array<any>> = [[]];
+    let section: Array<any> = [];
+    data.forEach(element => {
```

Comment:
Kindly consider not using *element*. Its not descriptive enough in the context of what the function does. Rename to something like *rowSection*